sklearn/preprocessing/label.py

```diff
@@ -54,15 +84,45 @@ def _encode_numpy(values, uniques=None, encode=False, check_unknown=True):
         return uniques
 
 
-def _encode_python(values, uniques=None, encode=False):
+class _TableWithNan(object):
+    #  hash table which allows nan as a key
+
+    def __init__(self):
+        self.dict = dict()
+        self.nan_value = None
+
+    def get(self, key):
```

Comment:
It seems strange to not do this with `__getitem__`. Indeed you could be inheriting from dict

Reply:
Yes indeed. Code updated.